README.md

```diff
@@ -50,73 +58,102 @@ If you want to stick with browserify, but use the Hot Module Reloading API (like
+### Example Installation (with `react-hot-loader` and Babel 6.x)
 
-Install pre-requirements (if not already exist)
+Install npm modules:
 
 ```sh
-npm i --save react
-npm i --save-dev watchify
+npm i --save react react-dom react-hot-loader@next babelify babel-preset-es2015 babel-preset-react
+npm i --save-dev watchify livereactload
 ```
 
-Install `babelify` and its dependencies
-
-```sh
-npm i --save babelify babel-preset-es2015 babel-preset-react
-```
-
-Install React proxying components and LiveReactload
-
-```sh
-npm i --save-dev livereactload react-proxy@1.x babel-plugin-react-transform
-```
-
-Create `.babelrc` file into project's root directory (or add `react-transform` extra
-if the file already exists). More information about `.babelrc` format and options
-can be found from [babel-plugin-react-transform](https://github.com/gaearon/babel-plugin-react-transform).
+Create a `.babelrc` file in the project's root directory (or add `react-hot-loader/babel` to the plugins section
+if the file already exists).
 
 ```javascript
 {
   "presets": ["es2015", "react"],
-  "env": {
-    "development": {
-      "plugins": [
-        ["react-transform", {
-          "transforms": [{
-            "transform": "livereactload/babel-transform",
-            "imports": ["react"]
-          }]
-        }]
-      ]
-    }
-  }
+  "plugins": ["react-hot-loader/babel"]
 }
 ```
 
-And finally use LiveReactload as a Browserify plugin with `watchify`. For example:
+Use LiveReactload as a Browserify plugin with `watchify`. For example:
 
 ```bash
 node_modules/.bin/watchify site.js -t babelify -p livereactload -o static/bundle.js
 ```
 
+Finally, when you create your application, the file where you render the top of the tree
+should look something like this:
+
+```javascript
+import React from 'react'
+import { render } from 'react-dom'
+import { AppContainer } from 'react-hot-loader'
+import Component from './components/Component'
+
+render(
+  <AppContainer>
+    <Component />
+  </AppContainer>,
+  document.getElementById('app')
+)
+```
+
 **That's it!** Now just start (live) coding! For more detailed example, please see
 **[the basic usage example](examples/01-basic-usage)**.
 
+**NOTE:** The above snippet is idempotent, which means it can be run over and over without issue. Sometimes
+you may place code here (like `redux` or `react-tap-event-plugin`) which are stateful and cannot be run over
+and over. In this case, you must use `module.hot.onUpdate` to indicate to LiveReactload what to do when a change
+is detected. See the []non-idempotent example](examples/04-non-idempotent).
+
+```javascript
+// index.js
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { AppContainer } from 'react-hot-loader'
+import MyComponent from './components/MyComponent'
+import getStore from './getStore'
+
+ReactDOM.render(
+  <AppContainer>
+    <MyComponent />
+  </AppContainer>,
+  document.getElementById('app')
+)
+
+// getStore.js
```

Comment:
Can we split these to two separate code blocks? 

In addition, I'm not sure what is the meaning of this code example because `module.hot.onUpdate` is not used even though it's mentioned above? If the purpose is to demonstrate non-idempotent example, could that be mentioned before the snipped?